packages/frontend/src/pages/campaigns/index.tsx

```diff
@@ -66,16 +68,25 @@ const STATE_OPTIONS = [
         label: t("stateOptions.expired"),
         value: CampaignState.EXPIRED,
     },
+    {
+        label: t("stateOptions.finalized"),
+        value: CampaignState.FINALIZED,
+    },
 ];
 
 export const Campaigns = () => {
     useResetPageScroll();
     const { t } = useTranslation();
 
     //  fetch KPITokens
-    const [results, setResults] = useState<KPIToken[]>([]);
-    const { searchQuery, setSearchQuery } = useSearch();
-    const { loading, kpiTokens } = useKPITokens(searchQuery);
+    const [results, setResults] = useState<ResolvedKPIToken[]>([]);
+    const [searchQuery, setSearchQuery] = useState("");
+    const [debouncedSearchQuery, setDebouncedSearchQuery] = useState("");
+
+    useDebounce(() => setDebouncedSearchQuery(searchQuery), 300, [searchQuery]);
```

Comment:
why do you prefer to declare the useDebounce logic explicitly on the component, instead of just using a hook like `useSearch` that hides the underlying logic? 

I think it's much simpler just using this hook.

`const { searchQuery, setSearchQuery } = useSearch();`

And I if I'm working on this component I don't need to care about the debounce for search if not working on it and it's easier to focus if you have less things to care about.

Reply:
It has literally no advantage at all. I mean, declaring a hook that ONLY calls one hook internally rarely makes sense unless there's some things that you want to make transparent to the user. But in this case I think it's overall clearer what this does compared to a `useSearch` (which also has a misleading name, because it has no search-specific logic and is just a hook to return a debounced value).